libs/ledger-live-common/src/families/bitcoin/hw-signMessage.ts

```diff
@@ -1,10 +1,14 @@
 import Btc from "@ledgerhq/hw-app-btc";
 import type { SignMessage } from "../../hw/signMessage/types";
 
-const signMessage: SignMessage = async (transport, { path, message }) => {
+const signMessage: SignMessage = async (transport, account, { message }) => {
+  if (typeof message !== "string") {
+    throw new Error("Invalid message type");
+  }
+
   const btc = new Btc({ transport });
-  const hexMessage = Buffer.from(message as string).toString("hex");
-  const result = await btc.signMessage(path, hexMessage);
+  const hexMessage = Buffer.from(message).toString("hex");
```

Comment:
Yup (if you're not talking about the removal of the casting), all chains are receiving hexa strings as of today

Reply:
What I meant is that for at least BTC you do
1. hex > utf8
2. utf8 > hex

So my point was to just keep the hex format and forward it to families, who may or may not format it differently.
But nbd really.

Reply:
Ah yes ! But that didn't change from the previous behaviour, and that can be explained by the fact that the flow handles both the UI expectatons and signature expectations. 
Every message is going trough the global `prepareMessageToSign` from [`hw/signMessage`](https://github.com/LedgerHQ/ledger-live/blob/aab36ad47ae95b1517308de686e3022634261297/libs/ledger-live-common/src/hw/signMessage/index.ts#L15) and its return is the one displayed on the clients, meaning that if you keep it as an hex string here, all clients will display the hex string as well unless you make them convert it.

So I guess we have multiple options:
- Option 1
  - Conversion is handled by the global `prepareMessageToSign` & returns utf8 from hex 
  -  `signMessage` using the return of `prepareMessageToSign` converts *when needed* to the right format (hex) to sign (for example EVM EIP712 message, or even cosmos might not need that as you sign JSON files) 
  - Clients using the return of `prepareMessageToSign` don't need to handle anything
 
- Option 2
  - Conversion is handled by the family specific `prepareMessageToSign` if needed & returns the original message (hex)
  - `signMessage` using the return of `prepareMessageToSign` doesn't need to convert if it's expecting hex string (but will need if it requires something else like for EIP712 signature where it needs a JSON, or the example of cosmos with JSON as well)
  - Clients using the return of `prepareMessageToSign` need to convert to utf8 to be able to display the message

You also have an Option 3 where every specific `prepareMessageToSign` returns its own potential conversion and the Clients & Signers need to check a `messageType` key maybe to know exactly how to act with it, but it won't change the fact that between the 3 actors `prepare`, `sign` & UI, conversions will be done somehow 🤷‍♂️

Reply:
Thanks for the details, doesn't matter a lot indeed since as you said conversion has to be done at some point, and the fact is that it will depend on each chain whether they use hex or utf8, so no option is clearly better than another.

LGTM :+1: